coalib/results/TextPosition.py

```diff
@@ -15,12 +19,21 @@ def __init__(self, line: (int, None)=None, column: (int, None)=None):
         :param line:        The line in file or None, the first line is 1.
         :param column:      The column indicating the character. The first one
                             in a line is 1.
+        :param text:        The text containing the character.
+        :param position:    Position identifying the index of character in text.
         :raises TypeError:  Raised when line or columns are no integers.
         :raises ValueError: Raised when a column is set but line is None.
         """
         if line is None and column is not None:
             raise ValueError("A column can only be set if a line is set.")
 
+        if line and column and text and not position:
+            position = TextPosition.calc_position(text, line, column)
+
+        if (line and column) is None and (text and position) is not None:
+            line, column = TextPosition.calc_line_col(text, position)
```

Comment:
@AbdealiJK  i do get it, i just want to be certain that this is the best way, so right now what you suggest is that we call `calc_position` when a user wants to use the `position` property, but what if an uknowing user makes multiple calls to the position property? it would call `calc_position` multiple times and would be a huge overhead, wouldn't calling it once if the text is specified better? it would only get called when the text is specified, meaning the user does wan't to use it, also sometimes the user might want to just store a position without the text, do we not consider those cases?